worker/manager/src/mirrors_qa_manager/worker.py

```diff
@@ -0,0 +1,309 @@
+        }
+        return query_backend_api(
+            endpoint,
+            method,
+            headers=req_headers,
+            payload=payload,
+        )
+
+    def fetch_tests(self) -> list[dict[str, Any]]:
+        logger.debug("Fetching tasks from backend API")
+
+        # Fetch tasks that were assigned to the worker that haven't been expired
+        params = urlencode({"worker_id": self.worker_id, "status": "PENDING"})
+        data = self.query_api(f"/tests?{params}")
+
+        logger.info(f"Fetched {data['metadata']['page_size']} test(s) from Backend API")
+
+        return data["tests"]
+
+    def sleep(self) -> None:
+        logger.info(f"Sleeping for {Settings.SLEEP_TIMEOUT}s")
+        time.sleep(Settings.SLEEP_TIMEOUT)
+
+    def run(self) -> None:
+        logger.info("Starting worker manager.")
+        try:
+            # Ensure the wireguard container is still up
+            exec_command(self.wg_container, self.wg_healthcheck_cmd)
+            while True:
+                tests = self.fetch_tests()
+                for test in tests:
+                    test_id = test["id"]
+                    country_code = test["country_code"]
+                    # Fetch the configuration file for the requested country
+                    try:
+                        self.copy_wireguard_conf_file(country_code)
+                    except FileNotFoundError:
+                        logger.warning(
+                            f"Could not find {country_code}.conf for "
+                            f"{test_id}. Skipping."
+                        )
+                        continue
+                    except Exception:
+                        logger.exception(
+                            f"error while fetching {country_code}.conf for {test_id}"
+                        )
+                        continue
+
+                    logger.info(
+                        f"Reconfiguring wireguard network interface for {country_code}"
+                    )
+
+                    # After copying the file, restart the interface.
+                    exec_command(self.wg_container, self.wg_down_cmd)
+                    exec_command(self.wg_container, self.wg_up_cmd)
+                    # Perform another healthcheck to ensure traffic can go
+                    # through.
+                    # TODO: Use the result from the healthcheck call to
+                    # populate the IP-related data as the task container
+                    # doesn't know anything about its network.
+                    # Could also be used to validate that the country config
+                    # is actually for this country in case the 'host'
+                    # wrongly names a config file.
+                    exec_command(self.wg_container, self.wg_healthcheck_cmd)
+
+                    logger.info(f"Processing test {test_id}")
+                    # Start container for the task
+                    output_filename = f"{test_id}.json"
```

Comment:
You can only define `output_fpath = self.instance_dir / f"{test_id}.json"` and when required use `output_fpath.name` to get filename